corgi/api/views_search.py

```diff
@@ -0,0 +1,129 @@
+import logging
+from urllib.parse import quote, unquote
+
+from django.db.models import Q
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from corgi.api.constants import CORGI_API_VERSION
+from corgi.core.models import Component
+
+logger = logging.getLogger(__name__)
+
+
+class SearchDeptopiaView(APIView):
+    """ """
+
+    def get(self, request, *args, **kwargs):
+        """
+        Loop through match components based on search terms
+        and return all sources (ancestors), providing product_stream.
+        """
+
+        name = None
+        if request.query_params.get("name"):
+            name = unquote(request.query_params.get("name"))
+
+        re_name = None
+        if request.query_params.get("re_name"):
+            re_name = unquote(request.query_params.get("re_name"))
+
+        purl = None
+        if request.query_params.get("purl"):
+            purl = unquote(request.query_params.get("purl"))
+
+        nevra = request.query_params.get("nevra")
+        nvr = request.query_params.get("nvr")
+
+        ecosystem = request.query_params.get("ecosystem")
+        q = request.query_params.get("q")
+
+        flatten = request.query_params.get("flatten")
+
+        # match_root = True
+        # match_dep = True
+        # if request.query_params.get("match_root"):
+        #     match_root = True
+        #     match_dep = False
+        # if request.query_params.get("match_dep"):
+        #     match_root = False
+        #     match_dep = True
+
+        query = Q()
+        if name:
+            query |= Q(name=name)
```

Comment:
I think instead of searching the name field directly here we should emulate the deptopia_cli behaviour which is to add a wildcard to the start of the search term, and anchor it to the end, eg:

```
Q(name__iendswith=name)
```

Compare the results with a search term such as `yaml`